VirtualList/tests/VirtualList-translate-specs.js

```diff
@@ -0,0 +1,810 @@
+					scrollMode="translate"
+				/>
+			);
+
+			const list = screen.getByRole('list');
+			fireEvent.wheel(list, {deltaY: 100});
+
+			act(() => jest.advanceTimersByTime(1000)); // Wait onScrollStop
+
+			expect(fn).toBeCalled();
+		});
+
+		test('should show overscroll effect by wheel', (done) => {
+			const fn = jest.fn();
+
+			const onScrollStop = handlerOnScrollStop(done, () => {
+				fn();
+				expect(startScrollTop).toBe(0);
+				expect(onScrollStartCount).toBe(1);
+				expect(resultScrollTop).toBe(0);
+
+				const listRoot = screen.getByRole('list').parentElement;
+				expect(listRoot).toHaveStyle({'--scroll-overscroll-translate-vertical': 'translateY(0px)'});
+			});
+
+			render(
+				<VirtualList
+					clientSize={clientSize}
+					dataSize={dataSize}
+					itemRenderer={renderItem}
+					itemSize={itemSize}
+					onScrollStop={onScrollStop}
+					onScrollStart={handlerOnScrollStart}
+					scrollMode="translate"
+				/>
+			);
+
+			const list = screen.getByRole('list');
+			fireEvent.wheel(list, {deltaY: -100});
+
+			act(() => jest.advanceTimersByTime(1000)); // Wait onScrollStop
+
+			expect(fn).toBeCalled();
+		});
+
+		test('should not scroll by wheel when \'noScrollByWheel\' prop is true', (done) => {
+			const fn = jest.fn();
+
+			render(
+				<VirtualList
+					clientSize={clientSize}
+					dataSize={dataSize}
+					itemRenderer={renderItem}
+					itemSize={itemSize}
+					noScrollByWheel
+					onScrollStop={fn}
+					onScrollStart={handlerOnScrollStart}
+					scrollMode="translate"
+				/>
+			);
+
+			const list = screen.getByRole('list');
+			fireEvent.wheel(list, {deltaY: 100});
+
+			act(() => jest.advanceTimersByTime(1000)); // Wait onScrollStop
+
+			expect(fn).not.toHaveBeenCalled();
+
+			done();
+		});
+	});
+
+	describe('Adding an item', () => {
+		test('should render an added item named \'Password 0\' as the first item', (done) => {
+			const itemArray = [{name: 'A'}, {name: 'B'}, {name: 'C'}];
+			const renderItemArray = ({index, ...rest}) => {
+				return (
+					<div {...rest} id={'item' + index}>
+						{itemArray[index].name}
+					</div>
+				);
+			};
+
+			const {rerender} = render(
+				<VirtualList
+					clientSize={clientSize}
+					dataSize={itemArray.length}
+					itemRenderer={renderItemArray}
+					itemSize={itemSize}
+					scrollMode="translate"
+				/>
+			);
+
+			itemArray.unshift({name: 'Password 0'});
+			rerender(<VirtualList
+				clientSize={clientSize}
+				dataSize={itemArray.length}
+				itemRenderer={renderItemArray}
+				itemSize={itemSize}
+				scrollMode="translate"
+			/>);
+
+			jest.useFakeTimers();
+
+			act(() => jest.advanceTimersByTime(0));
+			const expected = itemArray[0].name;
+			const actual = screen.getByRole('list').children.item(0).textContent;
+
+			expect(actual).toBe(expected);
+			done();
+
+			jest.useRealTimers();
+		});
+	});
+
+	describe('Voice Control', () => {
+		test('should render \'data-webos-voice-focused\' to outermost node of VirtualList', () => {
```

Comment:
a minor thing but would you please put 'the' in front of `outermost`s?

Reply:
I fixed it.